bakerydemo/base/middleware.py

```diff
@@ -0,0 +1,51 @@
+import re
+
+from django.conf import settings
+from django.core.exceptions import MiddlewareNotUsed
+from django.template.loader import render_to_string
+
+
+class DemoBannerMiddleware:
+    """
+    When bakerydemo is being run as the Wagtail Demo, show a banner at the top
+    of the page mentioning when the site will be reset.
+    """
```

Comment:
From what I understand this is something that would display at the top of all HTML responses regardless of the pages’ designs? This seems like too much to me – there are lots of places where we wouldn’t want such a disclaimer to start with, and even when we do, I don’t think we’d want to have a custom design for this that we’d need to keep supporting.

I think it’s very important the visual design of this disclaimer is as unobtrusive as possible, so we can get a sense of the visual design of Wagtail without distractions. My preference would be to either only have the disclaimer on the login screen in the existing `help_block` you’ve added, or login screen and in addition the Wagtail dashboard, also as a `help_block` that would naturally fit with Wagtail’s visual design.

Reply:
Given quite how brittle the current implementation is, I agree. Given changes can only be made after being logged in anyway, that'll be enough for people to see.